react/features/app/components/AbstractApp.js

```diff
@@ -25,6 +27,32 @@ import { appNavigate, appWillMount, appWillUnmount } from '../actions';
  */
 const DEFAULT_URL = 'https://meet.jit.si';
 
+/**
+ * react-redux's {@code Provider} component only accepts a single child, so use
+ * a simple wrapper component in order to pass more than 1 child components.
+ * FIXME: remove this once React Native supports Fragment.
+ */
+class AppRoot extends Component {
+    /**
+     * {@code AppRoot} component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        children: PropTypes.node
+    };
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return this.props.children;
+    }
+}
+
```

Comment:
Can you please do something about `AppRoot` because I'm totally not digging it?

1. The name `AppRoot` doesn't prepare me for what it actually does. (The comment is very good, of course, so that you very much for that!)
2. You added `AppRoot` because you needed a `Fragment` but then react-native didn't support it. Wouldn't such a `Fragment`-like component then be a good candidate for base/react because if you need it here, it may very well be a very good fit elsewhere (we probably have such uses already)? (And since you want a React `Fragment` for this purpose, maybe the react/base Component could be called `Fragment` instead of `AppRoot`? I'm not religious on `Fragment` because you could argue that we're shadowing React's abstraction and thus getting into two abstractions with one name territory but I'm not liking `AppRoot` at all.)

Reply:
You are absolutely right. I like the idea of calling it `Fragment`, since that's what we are going to replace it with.

I'm going to just rename it and leave a TODO comment, since the support for it is already in master, and will be part of RN 0.52, so it will happen soon and I don't think going all the way and adding it to base/react gives us an advantage here.

Reply:
> I'm going to just rename it and leave a TODO comment, since the support for it is already in master, and will be part of RN 0.52, so it will happen soon and I don't think going all the way and adding it to base/react gives us an advantage here

It took the combined effort of the three of us, more than a week (probably) to update us to .49. I did the update to .50 the other day. Now .51 is out and we haven't even planned to update to it. .52 is going to come out in January 2018 and I'm not sure when we'll have the time to do the update. I do not define that as "it will happen soon" and, moreover, it would've probably taken less time that the conversation that we're having right now.